addons/sale_management/models/sale_order_option.py

```diff
@@ -86,6 +88,8 @@ def _compute_price_unit(self):
             # To compute the price_unit a so line is created in cache
             values = option._get_values_to_add_to_order()
             new_sol = self.env['sale.order.line'].new(values)
+            if option.is_price_set_manually:
```

Comment:
you can do this directly in `_get_values_to_add_to_order` to avoid one extra write